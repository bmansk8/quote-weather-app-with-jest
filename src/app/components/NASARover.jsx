import React, { Component } from "react";
import getPhotos from './services/roverPhotos'

export const NASARover = class NASARover extends Component {
  constructor(props) {
    super(props);
    this.state = {
      imgs: []
    }
  }

  componentDidMount() {
    this.getImages();
  }

  async getImages() {
    const imgs = await getPhotos();
    this.setState({
      imgs: imgs
    })
  }



  render() {
    return (
      <div>
        <h1 id='roverH1' className='text-center '>Photos from mars</h1>
        <div id='rover-photos'>
          {this.state.imgs.map((obj, index) => (
            <img className='imgs shadow-lg rounded' src={obj.img_src} key={index}></img>
          ))}
        </div>
      </div>
    );
  }
}