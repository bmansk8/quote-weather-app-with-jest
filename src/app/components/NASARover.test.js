import React from "react";
import renderer from 'react-test-renderer';
import { NASARover } from './NASARover';
import { shallow } from 'enzyme'

it('renders h1 correctly', () => {
  const tree = renderer
    .create(<h1 className='text-center'>Photos from mars</h1>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})

jest.mock('./services/roverPhotos')

it('fetchs images from roverPhotos and renders them', done => {
  const wrapper = shallow(<NASARover />)

  setTimeout(()=>{
    wrapper.update();

    expect(wrapper.find('img').length).toEqual(3);

    done()
  });
  
});
