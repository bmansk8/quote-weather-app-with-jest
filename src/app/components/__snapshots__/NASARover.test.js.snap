// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders h1 correctly 1`] = `
<h1
  className="text-center"
>
  Photos from mars
</h1>
`;
